### PR TITLE
[17.04.x] deprecation and changelog updates

### PR DESCRIPTION
**Pull request by thaJeztah**
- Interacting with V1 registries is not yet disabled by default, so moving the deprecation version
- Added various entries to the changelog, and moved some to a different section

ping @vieux PTAL

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -68,9 +71,14 @@ be found.
 + Renew the context after communicating with the registry [#31586](https://github.com/docker/docker/pull/31586)
 
 ### Windows
+
 * Wait for OOBE to prevent crashing during host update [#31054](https://github.com/docker/docker/pull/31054)
 * Block pulling Windows images on non-Windows daemons [#29001](https://github.com/docker/docker/pull/29001)
 
+### Deprecation
+
+* Removal of the email from `docker login` [#26868](https://github.com/docker/docker/pull/26868)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Alternatively, we should revert this change in 17.04 and move it to 17.06

File: docs/deprecated.md [hunk 2]
```diff
@@ -266,7 +266,7 @@ of the `--changes` flag that allows to pass `Dockerfile` commands.
 
 ### Interacting with V1 registries
 
-**Disabled By Default In Release: v17.04**
+**Disabled By Default In Release: v17.05**
```
Comment (thaJeztah) [inline docs/deprecated.md]: Wondering if we should only deprecate in _quarterly_ releases, i.e. move this to 17.06? If so we should update the deprecation policy accordingly. We can _mark_ features deprecated in monthly releases, but _remove_ them only in quarterly releases wdyt?

Comment (cpuguy83) [inline docs/deprecated.md]: Yep.

Comment (thaJeztah) [inline docs/deprecated.md]: Let me move it to 17.06; we can decide on the change in policy separate from that

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,15 +26,23 @@ be found.
...
 + Add `.CreatedAt` placeholder for `docker network ls --format` [#29900](https://github.com/docker/docker/pull/29900)
 * Update order of `--secret-rm` and `--secret-add` [#29802](https://github.com/docker/docker/pull/29802)
-* Fix use of `**/` in `.dockerignore` [#29043](https://github.com/docker/docker/pull/29043)
 + Add `--filter enabled=true` for `docker plugin ls` [#28627](https://github.com/docker/docker/pull/28627)
 + Add `--format` to `docker service ls` [#28199](https://github.com/docker/docker/pull/28199)
 + Add `publish` and `expose` filter for `docker ps --filter` [#27557](https://github.com/docker/docker/pull/27557)
 * Support multiple service IDs on `docker service ps` [#25234](https://github.com/docker/docker/pull/25234)
 + Allow swarm join with `--availability=drain` [#24993](https://github.com/docker/docker/pull/24993)
-* Removal of the email from `docker login` [#26868](https://github.com/docker/docker/pull/26868)
++ Allow `plugin` management commands to address plugins by ID instead of name [#29222](https://github.com/docker/docker/pull/29222)
```
Comment (vieux) [inline CHANGELOG.md]: I removed this one already, as @anusha-ragunathan pointed out, the feature was already here, this PR only adds a test

File: CHANGELOG.md [hunk 4]
```diff
@@ -66,11 +84,18 @@ be found.
...
 + Renew the context after communicating with the registry [#31586](https://github.com/docker/docker/pull/31586)
++ (experimental) Add `--tail` and `--since` options to `docker service logs` [#31500](https://github.com/docker/docker/pull/31500)
++ (experimental) Add `--no-task-ids` and `--no-trunc` options to `docker service logs` [#31672](https://github.com/docker/docker/pull/31672)
 
 ### Windows
+
 * Wait for OOBE to prevent crashing during host update [#31054](https://github.com/docker/docker/pull/31054)
 * Block pulling Windows images on non-Windows daemons [#29001](https://github.com/docker/docker/pull/29001)
 
+### Deprecation
```
Comment (vieux) [inline CHANGELOG.md]: @tonistiigi pointed out in the last PR that it's usually used to announce new depreciation, not the actual removal.

File: CHANGELOG.md (hunk 1 above)
Comment (vieux) [inline CHANGELOG.md]: right, see https://github.com/docker/docker/pull/31862

File: CHANGELOG.md (hunk 3 above)
Comment (thaJeztah) [inline CHANGELOG.md]: oh, right, let me remove the impact/changelog label :)

File: CHANGELOG.md (hunk 4 above)
Comment (thaJeztah) [inline CHANGELOG.md]: hm, ok, wondering if it's better to have a separate section to more clearly point it out (it's easily overlooked and may be important when upgrading from a previous version)

open to suggestions though

**Comment by vieux** [comment]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM 🐯
